packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -0,0 +1,446 @@
+
+    this.activateDraw(evt, annotation, viewportIdsToRender);
+
+    evt.preventDefault();
+
+    triggerAnnotationRenderForViewportIds(renderingEngine, viewportIdsToRender);
+
+    return annotation;
+  };
+
+  /**
+   * Begins an edit of an open contour, when the mouse has selected a handle
+   * (end) of the open contour.
+   *
+   * @param evt - `EventTypes.MouseDownEventType`
+   * @param annotation - `PlanarFreehandROIAnnotation` annotation.
+   * @param handle - The handle index, 0 for the start and 1 for the end.
+   * @param interactionType - interaction type (mouse, touch)
+   */
+  handleSelectedCallback = (
+    evt: EventTypes.MouseDownEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    handle: ToolHandle,
+    interactionType = 'mouse'
+  ): void => {
+    const eventDetail = evt.detail;
+    const { element } = eventDetail;
+
+    const viewportIdsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName()
+    );
+
+    this.activateOpenContourEndEdit(evt, annotation, viewportIdsToRender);
+  };
+
+  /**
+   * Edits the open or closed contour when the line is grabbed and dragged.
+   */
+  toolSelectedCallback = (
+    evt: EventTypes.MouseDownEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    interactionType: InteractionTypes
+  ): void => {
+    const eventDetail = evt.detail;
+    const { element } = eventDetail;
+
+    const viewportIdsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName()
+    );
+
+    if (annotation.data.isOpenContour) {
+      this.activateOpenContourEdit(evt, annotation, viewportIdsToRender);
+    } else {
+      this.activateClosedContourEdit(evt, annotation, viewportIdsToRender);
+    }
+  };
+
+  /**
+   * Returns if the canvas point is near the line of the given annotation in the
+   * provided element or not. A proximity is passed to the function to determine the
+   * proximity of the point to the annotation in number of pixels.
+   *
+   * @param element - HTML Element
+   * @param annotation - The `PlanarFreehandROIAnnotation`.
+   * @param canvasCoords - Canvas coordinates
+   * @param proximity - Proximity to tool to consider
+   * @returns Boolean, whether the canvas point is near tool
+   */
+  isPointNearTool = (
+    element: HTMLDivElement,
+    annotation: PlanarFreehandROIAnnotation,
+    canvasCoords: Types.Point2,
+    proximity: number
+  ): boolean => {
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+
+    const points = annotation.data.polyline;
+
+    // NOTE: It is implemented this way so that we do not double calculate
+    // points when number crunching adjacent line segments.
+    let previousPoint = viewport.worldToCanvas(points[0]);
+
+    for (let i = 1; i < points.length; i++) {
+      const p1 = previousPoint;
+      const p2 = viewport.worldToCanvas(points[i]);
+
+      const distance = pointCanProjectOnLine(canvasCoords, p1, p2, proximity);
```

Comment:
`distance = pointCanProjectOnLine(` seems weird, since it is used later as a boolean check maybe do `canProject` or just `if (pointCanProject...)`